stablehlo/tests/ops_stablehlo_quantized.mlir

```diff
@@ -280,20 +584,48 @@ func.func @select_and_scatter_per_tensor_quantization(%arg0: tensor<10x24x24x64x
   func.return %0 : tensor<10x24x24x64x!quant.uniform<i8:f32, 1.0:17>>
 }
 
+// CHECK-QDQ-NOT: stablehlo.uniform_dequantize
+// CHECK-QDQ: stablehlo.select_and_scatter
+// CHECK-QDQ: stablehlo.uniform_dequantize
+// CHECK-QDQ: stablehlo.uniform_dequantize
+// CHECK-QDQ: stablehlo.compare
+// CHECK-QDQ-NOT: stablehlo.uniform_dequantize
+// CHECK-QDQ: stablehlo.add
+// CHECK-QDQ-NOT: stablehlo.uniform_quantize
```

Comment:
I'm confused by how these tests work..

```
// CHECK-QDQ-NOT: stablehlo.uniform_dequantize
// CHECK-QDQ: stablehlo.select_and_scatter
// CHECK-QDQ: stablehlo.uniform_dequantize
```

So it can't have a DQ op, but then after a select_and_scatter it can?

Also my thoughts on readability for most of the tests in this file:

1. Can we make better use of `CHECK-QDQ-NEXT:` to test locality to one another.
2. Can we add variables to enforce some graph structure:

```
// CHECK-QDQ-NEXT: %[[DQ_0:.*]] =  stablehlo.uniform_dequantize 
// CHECK-QDQ-NEXT: %[[DQ_1:.*]] =  stablehlo.uniform_dequantize 
...
// CHECK-QDQ-NEXT: stablehlo.compare %[[DQ_0]], %[[DQ_1]]
```

Reply:
Added context to each of the check.